### PR TITLE
Migrate to pyproject.toml, remove setup.cfg, pyscaffold

### PR DESCRIPTION
**Pull request by andmat900**
### Applicable Issues
- https://github.com/eiffel-community/etos/issues/208

### Description of the Change
This change removed setup.cfg and requirements.txt, migrating the package to pyproject.toml.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Andrei Matveyeu, andrei.matveyeu@axis.com

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,23 +1,42 @@
...
+    if version.distance == 0:
+        # If the current commit has a tag, do not add a local component, regardless of branch.
+        return ""
+    # For all other cases, always add the git reference (like "g7839952")
+    return f"+{version.node}"
 
 
 if __name__ == "__main__":
-    setup(use_pyscaffold=True)
+    setup()
```
Comment (t-persson) [inline setup.py:42]: Why not add local_scheme and version_scheme as you did the others?

Comment (andmat900) [reply setup.py]: Fixed now.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "etos_lib==4.3.1",
+    "opentelemetry-api~=1.21",
+    "opentelemetry-exporter-otlp~=1.21",
+    "opentelemetry-sdk~=1.21"
+]
+
+[options]
+zip_safe = false
+include_package_data = true
+python_requires = ">=3.4"
```
Comment (fredjn) [inline pyproject.toml:27]: Can we run with python 3.4?

Comment (andmat900) [reply pyproject.toml:27]: Most likely not. I've avoided changing the original value in this PR.

Comment (andmat900) [reply pyproject.toml]: Changed to >=3.9.

**Review by fredjn — APPROVED** [review]

**Pull request merged**